analysis_driver/pipelines/rapid.py

```diff
@@ -0,0 +1,238 @@
+                            'pc_duplicates': float(map_metrics['Number of duplicate reads (marked)'][1]),
+                            'mapped_reads': int(map_metrics['Mapped reads'][0]),
+                            'pc_mapped': float(map_metrics['Mapped reads'][1]),
+                            'unique_mapped_reads': int(map_metrics['Number of unique & mapped reads (excl. dups)'][0]),
+                            'pc_unique_mapped': float(map_metrics['Number of unique & mapped reads (excl. dups)'][1]),
+                            'mean_coverage': float(map_metrics['Average autosomal coverage'][0]),
+                            'median_coverage': float(map_metrics['Median autosomal coverage'][0]),
+                            'genome_size': int(map_metrics['Bases in reference genome'][0]),
+                            'pc_genome_with_coverage': {
+                                '100+': float(map_metrics['PCT of genome with coverage [100x:inf)'][0]),
+                                '50-100': float(map_metrics['PCT of genome with coverage [50x:100x)'][0]),
+                                '40-50': float(map_metrics['PCT of genome with coverage [40x:50x)'][0]),
+                                '30-40': float(map_metrics['PCT of genome with coverage [30x:40x)'][0]),
+                                '20-30': float(map_metrics['PCT of genome with coverage [20x:30x)'][0]),
+                                '10-20': float(map_metrics['PCT of genome with coverage [10x:20x)'][0]),
+                                '5-10': float(map_metrics['PCT of genome with coverage [ 5x:10x)'][0]),
+                                '2-5': float(map_metrics['PCT of genome with coverage [ 2x: 5x)'][0]),
+                                '1-2': float(map_metrics['PCT of genome with coverage [ 1x: 2x)'][0]),
+                                '0-1': float(map_metrics['PCT of genome with coverage [ 0x: 1x)'][0]),
+                            }
+                        }
+                    }
+                }
+            )
+
+        rest_communication.post_or_patch('samples', data, id_field='sample_id')
+        return 0
+
+
+class DragenOutput(RapidStage):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._datestamp = None
+
+    @property
+    def datestamp(self):
+        if self._datestamp is None:
+            self._datestamp = date.today().isoformat()
+        return self._datestamp
+
+    def delivery_folder(self, sample):
+        return os.path.join(
+            cfg['delivery']['dest'],
+            sample.project.name,
+            self.datestamp + '_rapid',
+            sample.udf['User Sample Name']
+        )
+
+    def _run(self):
+        exit_status = 0
+
+        for lane in sorted(self.dataset.rapid_samples_by_lane):
+            sample = self.dataset.rapid_samples_by_lane[lane]
+            sample_id = sample['sample_id']
+
+            if not self.review(sample_id):
```

Comment:
Why?

Reply:
`self.review` returns True if the sample passed QC, and we don't want to output/deliver failing samples.

Reply:
I see. A comment would be great here.